DirectProgramming/C++SYCL_FPGA/Tutorials/Features/IPAuthoring_interfaces_overview/mm-host/Readme.md

```diff
@@ -0,0 +1,238 @@
+# Memory-Mapped Host Interfaces
+This example design demonstrate a simple vector addition kernel that uses explicitly declared Memory-Mapped Host Interfaces to transfer data.
+
+This design will show how to explicitly declared Memory-Mapped Host Interfaces and how to customize their properties.
+
+## Invocation Interface
+By default, an un-decorated oneAPI kernel will have all its control signals and arguments passed through an MM-Agent interface.
+
+## Data Interface - MM Host
+By default, all the pointer members are implemented with MM Host interfaces. In this example design, interface `A_in`, `B_in`, `C_in` are explicitly declared with MM Host interfaces in order to customize their properties.
+
+You can customize memory-mapped interfaces of your IP component if the component uses a unified shared memory (USM) host pointer to access data. To customize the interface, use a functor to specify the component and declare your interfaces with templated type `annotated_arg` or `annotated_ptr`.
+
+`annotated_arg`/`annotated_ptr` support is enabled by including either of the following include statements in your design:
+```cpp
+#include <sycl/ext/oneapi/annotated_arg/annotated_arg.hpp>
+```
+Or
+```cpp
+#include <sycl/ext/intel/fpga_extensions.hpp>
+```
+
+### `annotated_arg` vs `annotated_ptr`
+You should use `annotated_ptr` for interfaces that are pointer functor members, and for other functor members, use `annotated_arg`. For detail, see [mmhost]() code sample.
+
+### Declare `annotated_arg`/`annotated_ptr`
+Explicit MM Host should be declared as functor member of your kernel:
+```cpp
+  sycl::ext::oneapi::experimental::annotated_arg<type, properies>
+
+  sycl::ext::oneapi::experimental::annotated_ptr<type_of_pointer, properies>
+```
+### MM Host Properties
+The `properties` template parameter is the type of a SYCL properties class object, including an unordered list of any of the compile-time properties listed in the following table.
```

Comment:
I don't think we need this table here; those properties will be discussed in the Avalon memory-mapped host tutorial and documentation.